src/fomo.jl

```diff
@@ -288,6 +289,9 @@ function SolverCore.solve!(
   solver.α = init_alpha(norm_∇fk, step_backend)
 
   # Stopping criterion: 
+  fmin = min(-one(T), f0) / eps(T)
```

Comment:
If you work in Float64, this isn’t an especially large number in absolute value. Float64 can represent values up to about 1.0e+308. It would be better to use `floatmax()` here. Effectively, when `f0 < -floatmax(Float64)`, it means that `f0 == Inf`. So you could declare unboundedness either if `f0 == Inf` or `f0 < floatmax(eltype(x)) / 2` (for example).